firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Driver/I2C/SFM3019/Device.txt

```diff
@@ -0,0 +1,253 @@
+
+    WHEN('The set_averaging function is called wirh averaging_window 128 as input parameter') 
+      THEN('The set_averaging function returns ok status') 
+      THEN('The mock I2C's write buffer consists of set_averaging  command byte '0x366A' followed by '0x0080FB'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+  GIVEN('A SFM3019 device constructed with a mock I2C device global I2c device of write buffer '0x01'  and air as gas parameter') 
+    WHEN('The set_averaging function is called wirh averaging_window 0 as input parameter') 
+      THEN('The set_averaging function returns ok status') 
+      THEN('The mock I2C's write buffer consists of set_averaging  command bytes '0x366A' followed by '0x000081'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+    WHEN('The set_averaging function is called wirh averaging_window 128 as input parameter') 
+      THEN('The set_averaging function returns ok status') 
+      THEN('The mock I2C's write buffer consists of set_averaging  command bytes '0x366A' followed by '0x0080FB'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+  GIVEN('A SFM3019 device constructed with a mock I2C device global I2c device of write buffer '0x01'  and mixture as gas parameter') 
+    WHEN('The set_averaging function is called wirh averaging_window 0 as input parameter') 
+      THEN('The set_averaging function returns ok status') 
+      THEN('The mock I2C's write buffer consists of set_averaging  command bytes '0x366A' followed by '0x000081'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+    WHEN('The set_averaging function is called wirh averaging_window 128 as input parameter') 
+      THEN('The set_averaging function returns ok status') 
+      THEN('The mock I2C's write buffer consists of set_averaging  command byte '0x366A' followed by '0x0080FB') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged ') 
+
+Scenario: SFM3019 Device:: SFM3019 device behaves correctly when request_conversion_factors function is called 
+  GIVEN('A SFM3019 device constructed with a mock I2C device , global I2C device has write buffer '0x01' and gas parameter') 
+    WHEN('The request_conversion_factors function is called with gas type O2 as input parameter') 
```

Comment:
Gas type as input parameter should be in the GIVEN - so we'll need 3 GIVENs in this scenario, with one WHEN for each GIVEN.